### DISCUSSION TITLE
How do I use the debugger?

### DISCUSSION THREAD

**Discussion by ghost** [NvChad/NvChad#1612] [Q&A]
Since it's supposed to be a functional IDE it must have a debugger but I haven't seen anything in the docs.

How do I use the debugger?

**Comment by siduck** [comment]
@JediMaster25 use any debugger plugin like nvim-dap with nvim-dap-ui

**Reply by janvhs** [reply]
Is there a particular reason for debugging not being part of the provided base configuration?

**Reply by siduck** [reply]
@bode-fun its always language specific and by default we support only lua lsp, so even if we were to add debugger config for lua in nvim-dap, you would have to override default config and then add your config, it'd be 1 step less if you just add it as your custom plugin

**Reply by janvhs** [reply]
Ahhh okay that makes sense to me... Maybe it would be a good idea to give a basic example for debugging in the docs... I would write some docs in a few weeks after I figured it out myself :)

**Reply by siduck** [reply]
you are right, im currently rewriting the docs https://github.com/NvChad/nvchad.github.io/pull/118 and improving its design & refactoring its code, after this i'll add an "extras" section where you'll find all sorts of cool neovim tricks and tips :D
